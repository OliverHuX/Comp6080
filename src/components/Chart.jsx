import React from 'react';
import { useTheme } from '@material-ui/core/styles';
import { LineChart, Line, XAxis, YAxis, ResponsiveContainer } from 'recharts';
import PropTypes from 'prop-types';
import Title from './Title';

function createData (question, amount) {
  return { question, amount };
}

export default function Chart ({ title, data }) {
  const theme = useTheme();
  return (
    <React.Fragment>
      <Title>{title}</Title>
      <ResponsiveContainer>
        {data && (<LineChart
          data={Array.from(data).map((val, index) => createData('Q' + index, val))}
          margin={{
            top: 16,
            right: 16,
            bottom: 0,
            left: 0,
          }}
        >
          <XAxis dataKey="question" stroke={theme.palette.text.secondary} />
          <YAxis stroke={theme.palette.text.secondary}>
          </YAxis>
          <Line type="monotone" dataKey="amount" stroke={theme.palette.primary.main} dot={false} />
        </LineChart>)}
      </ResponsiveContainer>
    </React.Fragment>
  );
}
Chart.propTypes = {
  title: PropTypes.string,
  data: PropTypes.array,
};
