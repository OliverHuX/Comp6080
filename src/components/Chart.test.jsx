import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Chart from './Chart';
import Title from './Title';

describe('ResultTable Component', () => {
  const data = [100, 30, 90, 80, 20, 100, 60];
  const title = 'sample title';
  it('uses the given title', () => {
    const chart = shallow(<Chart title={title} />);
    chart.contains(<Title>{title}</Title>);
  });
  it('has the given data in the chart', () => {
    const chart = shallow(<Chart data={data} title={title}/>);
    const newData = chart.childAt(1).childAt(0).props().data;
    expect(newData).toStrictEqual([
      { question: 'Q0', amount: 100 },
      { question: 'Q1', amount: 30 },
      { question: 'Q2', amount: 90 },
      { question: 'Q3', amount: 80 },
      { question: 'Q4', amount: 20 },
      { question: 'Q5', amount: 100 },
      { question: 'Q6', amount: 60 }
    ])
  });
  it('has data key question at x axis', () => {
    const chart = shallow(<Chart data={data} title={title}/>);
    const xAxis = chart.childAt(1).childAt(0).props().children[0];
    expect(xAxis.type.displayName).toBe('XAxis');
    expect(xAxis.props.dataKey).toBe('question');
  });
  it('has data key question at y axis', () => {
    const chart = shallow(<Chart data={data} title={title}/>);
    const yAxis = chart.childAt(1).childAt(0).props().children[2];
    expect(yAxis.type.displayName).toBe('Line');
    expect(yAxis.props.dataKey).toBe('amount');
  });
  it('renders with minimal props (snapshot)', () => {
    const chart = renderer.create(<Chart/>).toJSON();
    expect(chart).toMatchSnapshot();
  });
  it('renders with title props (snapshot)', () => {
    const chart = renderer.create(<Chart title={title} />).toJSON();
    expect(chart).toMatchSnapshot();
  });
  it('renders with title and data props (snapshot)', () => {
    const chart = renderer.create(<Chart title={title} data={data} />).toJSON();
    expect(chart).toMatchSnapshot();
  });
})
