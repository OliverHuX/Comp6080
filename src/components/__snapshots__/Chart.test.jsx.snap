// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultTable Component renders with minimal props (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  />,
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />,
]
`;

exports[`ResultTable Component renders with title and data props (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    sample title
  </h2>,
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />,
]
`;

exports[`ResultTable Component renders with title props (snapshot) 1`] = `
Array [
  <h2
    className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    sample title
  </h2>,
  <div
    className="recharts-responsive-container"
    style={
      Object {
        "height": "100%",
        "maxHeight": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "width": "100%",
      }
    }
  />,
]
`;
